Blogger Launchpad - add add_first_subscribers and verify_email tasks.

Addison-Stavlo opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Fixes # Automattic/loop#252 

related to https://github.com/Automattic/wp-calypso/pull/99207

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Adds a `add_first_subscribers` launchpad task definition. 
    * This task is intended to be marked complete when engaged with on the front-end, and not to depend on whether or not subscribers were actually added (noted in Automattic/loop#253). Because of this, we don't want to use or map to the `subscribers_added` task which requires adding subscribers for completion, but we check other subscribers tasks in the completion callback for this new task. (e.g. `subscribers_added` completion implies `add_first_subscribers` completion, but not the other way around).
* Adds the `add_first_subscribers` task and the `verify_email` task to the `write` intent launchpad task list.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Patch this branch to your sandbox: `bin/jetpack-downloader test jetpack-mu-wpcom-plugin add/subscribers-task-to-blogger-launchpad`  (taken from [instructions below](https://github.com/Automattic/jetpack/pull/41502#issuecomment-2630735041))
* Sandbox the public-api.
* Test alongside instructions in https://github.com/Automattic/wp-calypso/pull/99207

Addison-Stavlo committed:
add subscribers modal task to write

Addison-Stavlo committed:
add changelog

Addison-Stavlo committed:
add add_first_subscribers task to write intent

Addison-Stavlo committed:
update changelog

Addison-Stavlo committed:
dont map add_first_subscribers to the subscribers_added task

artemiomorales reviewed (dismissed):
Tests well and the code changes looks reasonable to me 👍

Addison-Stavlo committed:
add verify email task to blogger launchpad tasks

Addison-Stavlo committed:
add custom completion callback for add_first_subscribers task

Addison-Stavlo committed:
update changelog

Addison-Stavlo committed:
fix typo in changelog

Addison-Stavlo committed:
update comment for new task definition

Addison-Stavlo commented:
Since the last review I have:
* Added `wpcom_launchpad_is_add_first_subscribers_completed` callback to the new task
* Added the `verify_email` task to the write flow

DustyReagan reviewed (approved):
I tested this along with the Calypso PR, and it's working as expected. The comments are helpful as well.
